Fix unstable data-workspace tests

kisantia opened the pull request:
This removes the @UNSTABLE@ from the tests marked as unstable in #13822. I wasn't able to repro the failures locally, but I suspect the tests were flaky because they were creating a file and the validation for it existing wasn't always working. The changes in this PR: 
- stub the file existing function instead of actually creating files and checking if they exist
- change the validation tests to only call new workspace validation instead of overall validation, since the tests are only meant to validation that anyway

Charles-Gagnon commented on extensions/data-workspace/src/test/dialogs/newProjectDialog.test.ts:
```diff
@@ -48,25 +53,27 @@ suite('New Project Dialog', function (): void {
 		dialog.model.location = os.tmpdir();
 		dialog.workspaceInputBox!.value = 'test';
-		should.equal(await dialog.validate(), false, 'Validation should fail because workspace does not end in .code-workspace');
+		should.equal(await dialog.validateNewWorkspace(false), false, 'Validation should fail because workspace does not end in .code-workspace');
```
Why are you doing this instead of calling validate and then stubbing out whatever else validate is calling? That would make the most sense - by calling this function directly you're bypassing some of the normal logic which doesn't seem right

kisantia replied:
there was a separate test for the other validation in validate(), but I switched this back to to call validate instead of the new workspace validation

Charles-Gagnon commented on extensions/data-workspace/src/test/dialogs/newProjectDialog.test.ts:
```diff
@@ -48,25 +53,27 @@ suite('New Project Dialog', function (): void {
 		dialog.model.location = os.tmpdir();
 		dialog.workspaceInputBox!.value = 'test';
-		should.equal(await dialog.validate(), false, 'Validation should fail because workspace does not end in .code-workspace');
+		should.equal(await dialog.validateNewWorkspace(false), false, 'Validation should fail because workspace does not end in .code-workspace');
```
FYI - while looking at this code I noticed that dialogBase is implementing validate to always return true. It seems to me that the better thing would be to have validate be an abstract function instead so you're sure each page is implementing it correctly

kisantia replied:
updated it to be abstract

Charles-Gagnon reviewed (commented):
It would be nice to make it so the tests actually say what was wrong. The validate function does a couple of things and so it's impossible to tell exactly what went wrong.

I'm not sure why the file exists call would be failing either if the file/folder already exists. We write/read stuff all the time in tests and generally I would expect it to be fine.

alanrenmsft reviewed (approved):
in these unit test, we shouldn't have external dependencies, please double check whether other tests have such problems.

Charles-Gagnon commented:
FWIW I don't think we should consider local file system access an "external dependency". After all the app itself is running from that...

kisantia commented:
it was a little silly having almost identical tests for both dialogs, so I moved them to a separate file for DialogBase functions since the validateNewWorkspace is in DialogBase. Also added more checks to validate the error messages and split the test into one per scenario to make debugging easier in the future.
